lib/msal-core/src/UserAgentApplication.ts

```diff
@@ -1453,15 +1433,10 @@ protected getCachedTokenInternal(scopes : Array<string> , user: User): CacheResu
     else {
         acquireTokenUserKey = Constants.acquireTokenUser + Constants.resourceDelimeter + Constants.no_user + Constants.resourceDelimeter + authenticationRequest.state;
     }
-    if (Utils.isEmpty(this._cacheStorage.getItem(acquireTokenUserKey))) {
-      this._cacheStorage.setItem(acquireTokenUserKey, JSON.stringify(user));
-    }
-
+    
+    this._cacheStorage.setItem(acquireTokenUserKey, JSON.stringify(user));
     const authorityKey = Constants.authority + Constants.resourceDelimeter + authenticationRequest.state;
-    if (Utils.isEmpty(this._cacheStorage.getItem(authorityKey))) {
-      this._cacheStorage.setItem(authorityKey, authenticationRequest.authority);
-    }
-
+    this._cacheStorage.setItem(authorityKey, authenticationRequest.authority);
```

Comment:
are we missing the storeAuthStateInCookie flag here? You are passing it on line 412.

Reply:
same comment as above.